packages/cli/src/cli.ts

```diff
@@ -8,128 +8,128 @@ import { setLogLevel, Levels } from '@bigtest/logging';
-
-          console.log('Starting test run:', testRunId);
-
-          let subscription = yield client.subscribe(query.testRunResults(testRunId));
-
-          while(true) {
-            let { testRun } = yield subscription.receive();
-            if(testRun.status === 'ok') {
-              console.log('SUCCESS');
-              break;
-            }
-            if(testRun.status === 'failed') {
-              console.log('FAILED');
-              break;
-            }
-          };
-        });
-      })
-      .command('ci', 'start a server and run the test suite', (yargs) => {
-        yargs
-          .option('launch', {
-            describe: 'launch specified driver at server startup',
-            type: 'array',
-            default: []
-          })
-      }, (options) => {
-        setLogLevel(options.logLevel as Levels);
-
-        fork(function* server() {
-          let config: ProjectOptions = yield loadConfig(options.configFile as string | undefined);
-
-          let launch = options.launch as string[];
-          if (launch.length > 0) {
-            config.launch = launch;
-          }
-
-          for (let key of config.launch) {
-            if (!config.drivers[key]) {
-              throw new Error(`Could not find launch key ${key} in the set of drivers: ${JSON.stringify(Object.keys(config.drivers))}`);
-            }
-          }
-
-          let delegate = new Mailbox();
-          let atom = createOrchestratorAtom();
-          yield spawn(createOrchestrator({ atom, delegate, project: config }));
-
-          yield delegate.receive({ status: 'ready' });
-
-          let client: Client = yield Client.create(`ws://localhost:${config.port}`);
-
-          let { run: testRunId } = yield client.query(query.run());
-
-          console.log('Starting test run:', testRunId);
-
-          let subscription = yield client.subscribe(query.testRunResults(testRunId));
-
-          while(true) {
-            let { testRun } = yield subscription.receive();
-            if(testRun.status === 'ok') {
-              console.log('SUCCESS');
-              break;
-            }
-            if(testRun.status === 'failed') {
-              console.log('FAILED');
-              break;
-            }
-          };
-        });
-      })
-      .option('log-level', {
-        default: 'info',
-        global: true,
-        choices: ['debug', 'info', 'warn', 'error'],
-        desc: 'increase or decrease the amount of logging information printed to the console'
-      })
-      .option('config-file', {
-        global: true,
-        desc: 'path to the config file'
-      })
-      .demandCommand()
-      .help()
-      .parse(argv);
-    resume({})
-  };
+export function * CLI(argv: string[]): Operation {
+  let config: ProjectOptions = yield loadConfig();
+
+  let options = parseOptions(config, argv);
+
+  setLogLevel(options.logLevel);
+
+  if (options.command === 'server') {
```

Comment:
This is definitely something we could do in a follow-up PR as well, but it'd be pretty nice to extract the bodies of the branches of this if-statement into functions.

Reply:
100% agree